lib/graphql/static_validation/rules/argument_literals_are_compatible.rb

```diff
@@ -6,10 +6,15 @@ def validate_node(parent, node, defn, context)
         return if node.value.is_a?(GraphQL::Language::Nodes::VariableIdentifier)
         arg_defn = defn.arguments[node.name]
         return unless arg_defn
-        if !context.valid_literal?(node.value, arg_defn.type)
-          kind_of_node = node_type(parent)
-          error_arg_name = parent_name(parent, defn)
-          context.errors << message("Argument '#{node.name}' on #{kind_of_node} '#{error_arg_name}' has an invalid value. Expected type '#{arg_defn.type}'.", parent, context: context)
+
+        begin
+          if !context.valid_literal?(node.value, arg_defn.type)
+            kind_of_node = node_type(parent)
+            error_arg_name = parent_name(parent, defn)
+            context.errors << message("Argument '#{node.name}' on #{kind_of_node} '#{error_arg_name}' has an invalid value. Expected type '#{arg_defn.type}'.", parent, context: context)
+          end
+        rescue GraphQL::CoercionError => err
```

Comment:
this might be clearer if it *only* wrapped the `valid_literal?` call and then the default message was more obviously a fallback when no message is provided via `CoercionError`?